src/core/unit.c

```diff
@@ -6251,6 +6251,22 @@ bool unit_can_freeze(const Unit *u) {
         return UNIT_VTABLE(u)->freezer_action;
 }
 
+void unit_set_freezer_state(Unit *u, FreezerState state) {
+        assert(u);
+        assert(state >= 0);
+        assert(state < _FREEZER_STATE_MAX);
+
+        if (u->freezer_state == state)
+                return;
+
+        log_unit_debug(u, "Freezer state changed %s -> %s",
```

Comment:
very very very important: `special_glyph(SPECIAL_GLYPH_ARROW_RIGHT)`

Reply:
(I'd prefer not to use special glyphs in basic state change messages. None of the existing unit state change logs use this, and I think these should be readable even in the most limited environments)